django/db/models/fields/related.py

```diff
@@ -1051,6 +1106,50 @@ def _check_unique(self, **kwargs):
             else []
         )
 
+    @classmethod
+    def _has_related_models_with_db_cascading(cls, model, on_delete):
+        """
+        If the foreignkey parent has DB cascading and the Current model has non
+        db cascading return true
+        """
+        non_db_related_models = {}
+        # Optimization for the case when the model does not have non-db deletion
+        if not hasattr(model, "_meta") or on_delete in [
+            DB_CASCADE,
+            DB_SET_NULL,
+            DB_RESTRICT,
+        ]:
+            return non_db_related_models
+        # Fetch all the models related to the current model
+        # In other words fetch all the foreignkey childs.
+        related_models = [
+            rel.related_model
+            for rel in model._meta.get_fields()
+            if rel.related_model
+            and not rel.auto_created
+            and hasattr(rel.related_model, "_meta")
+        ]
+        for rel_model in related_models:
+            # check through the related models
+            # if they have DB level deletion return True
+            # Our current model already has non_db cascade
+            for rel in rel_model._meta.get_fields():
+                related_on_delete = None
+                related_remote_field = rel
+                if isinstance(rel, OneToOneRel) and hasattr(rel, "on_delete"):
+                    related_on_delete = rel.on_delete
+                    related_remote_field = rel.remote_field
+                elif (
+                    isinstance(rel, ForeignKey) or isinstance(rel, OneToOneField)
+                ) and hasattr(rel.remote_field, "on_delete"):
+                    related_on_delete = rel.remote_field.on_delete
+                if related_on_delete in [DB_CASCADE, DB_SET_NULL, DB_RESTRICT]:
+                    non_db_related_models["model"] = rel_model
+                    non_db_related_models["field"] = related_remote_field
+                    return non_db_related_models
+
+        return non_db_related_models
```

Comment:
It might be helpful to specify which parent model(s) (or `parent_ptr` field(s)) are incompatible. As the current field does not refers to the exact field on the parent model, it will be helpful for the user to know which parent model field uses non_db on_delete options.

Reply:
Earlier I made it with just a true/false flag, got some community review regarding the specification : https://github.com/django/django/pull/16851#discussion_r1213358420

Reply:
You always return one model and one field, so why `non_db_related_models` is a dictionary? It's enough to return a field and get model from it, check out my [comment](https://github.com/django/django/pull/16851#discussion_r1251632988).

Reply:
Got it, will update the PR soon.